src/scripts/main.js

```diff
@@ -1,3 +1,134 @@
 'use strict';
 
-// write code here
+function appendRow(button) {
+  const table = document.querySelector('.field > tbody');
+  const existRow = table.querySelector('tr');
+  const newRow = document.createElement('tr');
+
+  if (table.querySelectorAll('tr').length >= 10) {
+    button.disabled = true;
+
+    return 0;
+  }
+
+  newRow.innerHTML = existRow.innerHTML;
+
+  table.appendChild(newRow);
+
+  toggleButtons();
+}
+
+function removeRow(button) {
+  const table = document.querySelector('.field > tbody');
+  const rowToDel = table.querySelector('tr');
+
+  if (table.querySelectorAll('tr').length <= 2) {
+    button.disabled = true;
+
+    return 0;
+  }
+
+  rowToDel.remove();
+
+  toggleButtons();
+}
+
+function appendCol(button) {
+  const table = document.querySelector('.field > tbody');
+  const rows = [...table.querySelectorAll('tr')];
+
+  if (rows[0].querySelectorAll('td').length >= 10) {
+    button.disabled = true;
+
+    return 0;
+  }
+
+  rows.forEach((row) => {
+    const newCell = document.createElement('td');
+
+    row.appendChild(newCell);
+  });
+
+  toggleButtons();
+}
+
+function removeCol(button) {
+  const table = document.querySelector('.field > tbody');
+  const rows = [...table.querySelectorAll('tr')];
+
+  if (rows[0].querySelectorAll('td').length <= 2) {
+    button.disabled = true;
+
+    return 0;
+  }
+
+  rows.forEach((row) => {
+    const cellToDel = row.querySelector('td');
```

Comment:
The `removeCol` function always removes the first column. If the intention is to remove a specific column, you should modify the logic to target the correct column.